lib/kaguya/core.ex

```diff
@@ -128,6 +133,7 @@ defmodule Kaguya.Core do
     cancel_server_timer(state)
     socket = reconnect()
     send(self(), :init)
+    send(self(), :rejoin_chan)
```

Comment:
This should be fine, since `send` should guarantee ordering (see [this](http://erlang.org/pipermail/erlang-questions/2009-January/041272.html)). So, init gets received, and the core is told to send nick/user etc. then rejoin_chan is received and then the channel messages get sent.